src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -619,22 +619,59 @@ private String determineKfsOwnershipTypeCodeFromPmwTaxClassificationCode(Payment
         return kfsVendorDataWrapper;
     }
-    
+
     private KfsVendorDataWrapper populateFirstLastLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
         kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(true);
-        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(pmwVendor.getRequestingCompanyLegalFirstName());
-        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(pmwVendor.getRequestingCompanyLegalLastName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(truncateLegalLastNameToMaximumAllowedLength(pmwVendor.getRequestingCompanyLegalLastName()));
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(pmwVendor.getRequestingCompanyLegalLastName(), pmwVendor.getRequestingCompanyLegalFirstName()));
         return kfsVendorDataWrapper;
     }
 
+    private String truncateBusinessLegalNameToMaximumAllowedLength(String businessLegalName) {
+        if (businessLegalName.length() <= VendorConstants.MAX_VENDOR_NAME_LENGTH) {
+            return businessLegalName;
+        } else {
+            String truncatedBusinessLegalName = businessLegalName.substring(0, VendorConstants.MAX_VENDOR_NAME_LENGTH);
+            LOG.info("truncateBusinessLegalNameToMaximumAllowedLength: Received businessLegalName '" + businessLegalName
+                    + "' with length of " + businessLegalName.length() + " and it is being truncated to '"
+                    + truncatedBusinessLegalName + "' with length of " + truncatedBusinessLegalName.length());
+            return truncatedBusinessLegalName;
+        }
+    }
+
+    private static String truncateLegalLastNameToMaximumAllowedLength(String legalLastName) {
+        if (legalLastName.length() <= VendorConstants.MAX_VENDOR_NAME_LENGTH) {
+            return legalLastName;
+        } else {
+            String truncatedLegalLastName = legalLastName.substring(0, VendorConstants.MAX_VENDOR_NAME_LENGTH);
+            LOG.info("truncateBusinessLegalNameToMaximumAllowedLength: Received legalLastName '" + legalLastName
+                    + "' with length of " + legalLastName.length() + " and it is being truncated to '"
+                    + truncatedLegalLastName + "' with length of " + truncatedLegalLastName.length());
+            return truncatedLegalLastName;
+        }
+    }
+
+    private static String truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(String legalLastName, String legalFirstName) {
+        int maxAllowedLengthOfFirstName = VendorConstants.MAX_VENDOR_NAME_LENGTH - VendorConstants.NAME_DELIM.length() - legalLastName.length();
+        if (legalFirstName.length() <= maxAllowedLengthOfFirstName) {
+            return legalFirstName;
+        } else {
+            String truncatedLegalFirstName = legalFirstName.substring(0, maxAllowedLengthOfFirstName);
```

Comment:
If the length of the legal last name is already the max length, then the legal first name would get truncated to an empty string.  I think this code may handle it, but if you put in an explicit check for that, it may make the code intention more easily understood.

Reply:
Good catch.  I have reworked this section.
To help clarify: 45 is max length of legal name no matter how it is created. If first and last is used to create it then two of those characters need to be a comma followed by a space to delimit them. 
So: If last is at the max or would be at the max with the delimiter added, I agree we should set first to empty string OR return the actual first name if the whole thing can be concatenated OR  create a sub-string to use for first.

Reply:
that's a good solution, thanks